include/RAJA/policy/cuda/multi_reduce.hpp

```diff
@@ -451,10 +450,7 @@ struct MultiReduceGridAtomicHostInit_Data
   //! combine value on host, combine a value into the tally
   void combine_host(int bin, T value)
   {
-    int tally_rep = 0;
-#if defined(RAJA_ENABLE_OPENMP)
-    tally_rep = omp_get_thread_num();
-#endif
+    int tally_rep = RAJA::get_thread_num<ThreadPolicy>();
```

Comment:
Nice idea, if your file is not compiled with openmp you don't have to pay the overhead of supporting a tally large enough to support the number of OMP threads.
If you tried to pass one of these objects between files with different openmp activity then it would not compile due to the differing thread policies which is good. You shouldn't do that anyway, but if anyone ever does spotting that the types are actually different may not be trivial.